src/block/accordion/edit.js

```diff
@@ -171,42 +170,16 @@ export default compose(
 
 // Add another icon picker to the Icon block for picking the icon for the opened accordion.
 addFilter( 'stackable.block-component.icon.after', 'stackable/blockquote', output => {
-	const { clientId } = useBlockEditContext()
+	const { parentTree } = useBlockContext()
+	const { getBlock } = useSelect( 'core/block-editor' )
 	const { getActiveBlockVariation } = useSelect( 'core/blocks' )
+	const { getAttribute, updateAttributeHandler } = useAttributeEditHandlers()
 
-	const {
-		getAttribute,
-		updateAttributeHandler,
-	} = useAttributeEditHandlers()
-
-	const { isAccordionIcon, block } = useSelect(
-		select => {
-			const { getBlock } = select( 'core/block-editor' )
-			const { parentTree } = select( 'stackable/block-context' ).getBlockContext( clientId )
-			const columnClientId = nth( parentTree, -2 )?.clientId
-			const accordionClientId = nth( parentTree, -3 )?.clientId
-			const iconLabelName = nth( parentTree, -1 )?.name
-			const columnName = nth( parentTree, -2 )?.name
-			const accordionName = nth( parentTree, -3 )?.name
-			if ( ! iconLabelName || ! columnName || ! accordionName ) {
-				return false
-			}
-			if ( iconLabelName !== 'stackable/icon-label' ||
-			     columnName !== 'stackable/column' ||
-				 accordionName !== 'stackable/accordion' ) {
-				return false
-			}
-			return {
-				isAccordionIcon: ! getBlock( accordionClientId ).innerBlocks[ 0 ].clientId !== columnClientId,
-				block: getBlock( accordionClientId ),
-			}
-		},
-		[ clientId ]
-	)
+	const accordionBlock = parentTree.find( pt => pt.name === 'stackable/accordion' )
```

Comment:
Because this matches all the parents of the block, this will also affect other Icon pickers that are nested as the contents of the Accordion block. For example if you have an accordion block, then inside the actual accordion container, you have an icon block, this would also reset to a chevron.

You'll need to check specifically if the icon block belongs to the header part of the accordion block.